features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPoolTest.kt

```diff
@@ -358,6 +361,53 @@ internal class BitmapPoolTest {
         assertThat(captor.firstValue).isEqualTo(0)
     }
 
+    @RepeatedTest(3000)
+    fun `M not receive ConcurrentModificationException W get() { and onLowMemory }`() {
+        // when this occurs the frequency is roughly once per 3000 runs,
+
+        // Given
+        val threadList = mutableListOf<Thread>()
+        val numberOfThreads = 3
+        val allThreadsStartedLatch = CountDownLatch(numberOfThreads)
+        val allThreadsFinishedLatch = CountDownLatch(numberOfThreads)
+        testedCache.put(mockBitmap)
+        val bitmapKey = spyBitmapPoolHelper.generateKey(mockBitmap)
+        val failed = AtomicBoolean(false)
+
+        // When
+        repeat(numberOfThreads) {
+            if (Math.random() < 0.5) {
+                threadList.add(
+                Thread {
+                    allThreadsStartedLatch.countDown()
+                    allThreadsStartedLatch.await()
+                    try {
+                        testedCache.get(bitmapKey)
+                    } catch (e: ConcurrentModificationException) {
+                        failed.set(true)
+                    }
+                    allThreadsFinishedLatch.countDown()
+                }
+                )
+            } else {
+                threadList.add(
+                Thread {
+                    allThreadsStartedLatch.countDown()
+                    allThreadsStartedLatch.await()
+                    testedCache.onLowMemory()
+                    allThreadsFinishedLatch.countDown()
+                }
+                )
+            }
+        }
+
+        threadList.forEach { it.start() }
+        allThreadsFinishedLatch.await()
+
+        // Then
+        MatcherAssert.assertThat("Received ConcurrentModificationException when getting bitmap", !failed.get())
```

Comment:
maybe we can just do `assertThat(failed.get).overridingErrorMessage(...).isFalse` without `MatcherAssert` usage.